File: Node/pdf-with-added-image.js
import fetch, { FormData, fileFromSync } from "node-fetch";

let formdata = new FormData();
formdata.append("file", fileFromSync("../Sample_Input/toAddImage.pdf"));
formdata.append("image_file", fileFromSync("../Sample_Input/pdfRestApiLab.png"));
formdata.append("output", "example_out");
formdata.append("x", "200");
formdata.append("y", "650");
formdata.append("page", "1");

let requestOptions = {
  method: "POST",
  headers: {
    "Api-Key": "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx", // Place your api key here
  },
  body: formdata,
  redirect: "follow",
};

fetch("https://api.pdfrest.com/pdf-with-added-image", requestOptions)
  .then((response) => response.text())
  .then((result) => console.log(result))
  .catch((error) => console.log("error", error));

File: Python/pdf-with-added-image.py
from requests_toolbelt import MultipartEncoder
import requests
import json

pdf_with_added_image_endpoint_url = 'https://api.pdfrest.com/pdf-with-added-image'

mp_encoder_pdfWithAddedImage = MultipartEncoder(
    fields={
        'file': ('toAddImage.pdf', open('../Sample_Input/toAddImage.pdf', 'rb'), 'application/pdf'),
        'image_file': ('pdfRestApiLab.png', open('../Sample_Input/pdfRestApiLab.png', 'rb'), 'image/png'),
        'output' : 'example_out',
        'x' : '200',
        'y' : '650',
        'page' : '1',
    }
)

headers = {
    'Accept': 'application/json',
    'Content-Type': mp_encoder_pdfWithAddedImage.content_type,
    'Api-Key': 'xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx' # place your api key here
}

print("Sending POST request to pdf-with-added-image endpoint...")
response = requests.post(pdf_with_added_image_endpoint_url, data=mp_encoder_pdfWithAddedImage, headers=headers)

print("Response status code: " + str(response.status_code))

if response.ok:
    response_json = response.json()
    print(json.dumps(response_json, indent = 2))
else:
    print(response.text)